pkg/storage/tree/tree.go

```diff
@@ -181,6 +181,33 @@ func (t *Tree) Iterate(cb func(key []byte, val uint64)) {
 	}
 }
 
+func (t *Tree) Iterate2(cb func(name string, self uint64, stack []string)) { // TODO
+	t.RLock()
+	defer t.RUnlock()
```

Comment:
Note that in `Collapsed` method locking is done explicitly before calling `Iterate2` while in `Pprof` is not, better to make this consistent. Since `Iterate` already leaves locking to the caller, the same approach could be used here.